src/redis/commands.cr

```diff
@@ -1713,5 +1862,89 @@ class Redis
       concat(destination, source2)
       destination
     end
+
+    private def hashify(ary : Array(RedisValue))
+      Hash(String, RedisValue).new.tap do |hsh|
+        ary.in_groups_of(2) { |(k, v)| hsh[k.to_s] = v }
+      end
+    end
+
+    private def hashify_stream_entries(ary)
+      StreamMessagesValue.new.tap do |entries|
+        ary.each do |entry|
+          if entry.is_a? Array(Redis::RedisValue)
+            entries[entry.first.to_s] = array_to_string_hash entry.last
+          end
+        end
+      end
+    end
+
+    private def hashify_streams(ary)
+      StreamsValue.new.tap do |streams|
+        ary.each do |stream|
+          if stream.is_a? Array(Redis::RedisValue)
+            streams[stream.first.to_s] = hashify_stream_entries stream.last.as(Array(Redis::RedisValue))
+          end
+        end
+      end
+    end
+
+    private def hashify_stream_pendings(result)
+      hsh = Hash(String, String | Hash(String, String)).new
+      return hsh unless result.is_a? Array(Redis::RedisValue)
+
+      consumers = array_pairs_to_string_hash result[3]
+      hsh["size"] = result[0].to_s
+      hsh["min_entry_id"] = result[1].to_s
+      hsh["max_entry_id"] = result[2].to_s
+      hsh["consumers"] = consumers
+      hsh
+    end
+
+    private def hashify_stream_pending_details(result)
+      Array(Hash(String, String)).new.tap do |ary|
+        if result.is_a? Array(Redis::RedisValue)
+          result.each do |arr|
+            if arr.is_a? Array(RedisValue)
+              ary << {
```

Comment:
This should certainly be a struct

```crystal
struct StreamPendingDetails
  property entry_id : String
  property consumer : String
  property elapsed : String
  property count : String
  def initialize(result : Array(RedisValue)
    @entry_id = result[0].to_s
    @consumer = result[1].to_s
    @elapsed = result[2].to_s
    @count = result[3].to_s
  end
  def self.new_collection(result : Array(RedisValue)) : Array(StreamPendingDetails)
    Array(StreamPendingDetails).new.tap do |ary|
      result.each do |result_ary|
        ary << StreamPendingDetails.new result_ary
      end
    end
  end
end
```
There's no sense in using a hash where all potential keys are known.